From 815832ca35550cae0782a974fb3814db4c332c7e Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Mon, 14 May 2018 11:20:54 +0300
Subject: [PATCH 0775/1200] phy: comphy: add support for SFI1

In CP115, comphy4 can be configured into SFI port1
(in addition to SFI0). This patch adds the option
described above.

In addition, rename all existing SFI/XFI references:
COMPHY_TYPE_SFI --> COMPHY_TYPE_SFI0

No functionl change for exsiting configuration.

Change-Id: Ib2ba004d1d7c92330fed58b14585576bbf7c89d6
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/55029
Reviewed-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/57593
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
---
 arch/arm/dts/armada-7020-amc.dts         |  2 +-
 arch/arm/dts/armada-7040-db-C.dts        |  2 +-
 arch/arm/dts/armada-7040-db-D.dts        |  2 +-
 arch/arm/dts/armada-7040-db.dts          |  2 +-
 arch/arm/dts/armada-7040-pcac.dts        |  2 +-
 arch/arm/dts/armada-8040-db-C.dts        |  4 ++--
 arch/arm/dts/armada-8040-db-D.dts        |  4 ++--
 arch/arm/dts/armada-8040-db-E.dts        |  4 ++--
 arch/arm/dts/armada-8040-db-F.dts        |  2 +-
 arch/arm/dts/armada-8040-db.dts          |  4 ++--
 arch/arm/dts/armada-8040-mcbin.dts       |  4 ++--
 arch/arm/dts/armada-8040-ocp.dts         |  4 ++--
 drivers/phy/marvell/comphy_core.c        |  4 ++--
 drivers/phy/marvell/comphy_cp110.c       | 14 ++++++++------
 include/dt-bindings/comphy/comphy_data.h |  9 +++++----
 15 files changed, 33 insertions(+), 30 deletions(-)

diff --git a/arch/arm/dts/armada-7020-amc.dts b/arch/arm/dts/armada-7020-amc.dts
index f902d17459..e47284adf9 100644
--- a/arch/arm/dts/armada-7020-amc.dts
+++ b/arch/arm/dts/armada-7020-amc.dts
@@ -168,7 +168,7 @@
 	};
 
 	phy4 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 		phy-speed = <COMPHY_SPEED_10_3125G>;
 	};
 
diff --git a/arch/arm/dts/armada-7040-db-C.dts b/arch/arm/dts/armada-7040-db-C.dts
index 479ecd8fb0..500b49f905 100644
--- a/arch/arm/dts/armada-7040-db-C.dts
+++ b/arch/arm/dts/armada-7040-db-C.dts
@@ -152,7 +152,7 @@
 	};
 
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 
 	phy3 {
diff --git a/arch/arm/dts/armada-7040-db-D.dts b/arch/arm/dts/armada-7040-db-D.dts
index 34ce6af7d6..e07ae3e93b 100644
--- a/arch/arm/dts/armada-7040-db-D.dts
+++ b/arch/arm/dts/armada-7040-db-D.dts
@@ -134,7 +134,7 @@
 	};
 
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 		phy-speed = <COMPHY_SPEED_10_3125G>;
 	};
 
diff --git a/arch/arm/dts/armada-7040-db.dts b/arch/arm/dts/armada-7040-db.dts
index 4d8aa7bde9..f70db9e4e0 100644
--- a/arch/arm/dts/armada-7040-db.dts
+++ b/arch/arm/dts/armada-7040-db.dts
@@ -193,7 +193,7 @@
 	};
 
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 		phy-speed = <COMPHY_SPEED_10_3125G>;
 	};
 
diff --git a/arch/arm/dts/armada-7040-pcac.dts b/arch/arm/dts/armada-7040-pcac.dts
index ce0baa8ac3..2beaa73211 100644
--- a/arch/arm/dts/armada-7040-pcac.dts
+++ b/arch/arm/dts/armada-7040-pcac.dts
@@ -154,7 +154,7 @@
 	};
 
 	phy4 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 		phy-speed = <COMPHY_SPEED_10_3125G>;
 	};
 
diff --git a/arch/arm/dts/armada-8040-db-C.dts b/arch/arm/dts/armada-8040-db-C.dts
index 997fd6fd4d..3940e0ce0b 100644
--- a/arch/arm/dts/armada-8040-db-C.dts
+++ b/arch/arm/dts/armada-8040-db-C.dts
@@ -121,7 +121,7 @@
 		phy-type = <COMPHY_TYPE_SATA0>;
 	};
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 	phy3 {
 		phy-type = <COMPHY_TYPE_SATA1>;
@@ -249,7 +249,7 @@
 		phy-type = <COMPHY_TYPE_SATA0>;
 	};
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 	phy3 {
 		phy-type = <COMPHY_TYPE_SATA1>;
diff --git a/arch/arm/dts/armada-8040-db-D.dts b/arch/arm/dts/armada-8040-db-D.dts
index 5c308afe76..755e02dfdb 100644
--- a/arch/arm/dts/armada-8040-db-D.dts
+++ b/arch/arm/dts/armada-8040-db-D.dts
@@ -121,7 +121,7 @@
 		phy-type = <COMPHY_TYPE_SATA0>;
 	};
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 	phy3 {
 		phy-type = <COMPHY_TYPE_SATA1>;
@@ -251,7 +251,7 @@
 		phy-type = <COMPHY_TYPE_SATA0>;
 	};
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 	phy3 {
 		phy-type = <COMPHY_TYPE_SATA1>;
diff --git a/arch/arm/dts/armada-8040-db-E.dts b/arch/arm/dts/armada-8040-db-E.dts
index 9d38928ad7..96d96bd8f7 100644
--- a/arch/arm/dts/armada-8040-db-E.dts
+++ b/arch/arm/dts/armada-8040-db-E.dts
@@ -121,7 +121,7 @@
 		phy-type = <COMPHY_TYPE_SATA0>;
 	};
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 	phy3 {
 		phy-type = <COMPHY_TYPE_SATA1>;
@@ -233,7 +233,7 @@
 		phy-type = <COMPHY_TYPE_SATA0>;
 	};
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 	phy3 {
 		phy-type = <COMPHY_TYPE_SATA1>;
diff --git a/arch/arm/dts/armada-8040-db-F.dts b/arch/arm/dts/armada-8040-db-F.dts
index 4103b25f3c..a70d32f41e 100644
--- a/arch/arm/dts/armada-8040-db-F.dts
+++ b/arch/arm/dts/armada-8040-db-F.dts
@@ -296,7 +296,7 @@
 		phy-type = <COMPHY_TYPE_SATA0>;
 	};
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 	phy3 {
 		phy-type = <COMPHY_TYPE_SATA1>;
diff --git a/arch/arm/dts/armada-8040-db.dts b/arch/arm/dts/armada-8040-db.dts
index 52d080aaec..e7d263c2e0 100644
--- a/arch/arm/dts/armada-8040-db.dts
+++ b/arch/arm/dts/armada-8040-db.dts
@@ -158,7 +158,7 @@
 		phy-type = <COMPHY_TYPE_SATA0>;
 	};
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 	phy3 {
 		phy-type = <COMPHY_TYPE_SATA1>;
@@ -265,7 +265,7 @@
 		phy-type = <COMPHY_TYPE_SATA0>;
 	};
 	phy2 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 	phy3 {
 		phy-type = <COMPHY_TYPE_SATA1>;
diff --git a/arch/arm/dts/armada-8040-mcbin.dts b/arch/arm/dts/armada-8040-mcbin.dts
index a36510c2b2..e7de8e21dd 100644
--- a/arch/arm/dts/armada-8040-mcbin.dts
+++ b/arch/arm/dts/armada-8040-mcbin.dts
@@ -211,7 +211,7 @@
 		phy-type = <COMPHY_TYPE_PEX0>;
 	};
 	phy4 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 	phy5 {
 		phy-type = <COMPHY_TYPE_SATA1>;
@@ -357,7 +357,7 @@
 		phy-type = <COMPHY_TYPE_SATA1>;
 	};
 	phy4 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 	};
 	phy5 {
 		phy-type = <COMPHY_TYPE_SGMII2>;
diff --git a/arch/arm/dts/armada-8040-ocp.dts b/arch/arm/dts/armada-8040-ocp.dts
index 15b9619886..386e6dded9 100644
--- a/arch/arm/dts/armada-8040-ocp.dts
+++ b/arch/arm/dts/armada-8040-ocp.dts
@@ -172,7 +172,7 @@
 		phy-type = <COMPHY_TYPE_IGNORE>;
 	};
 	phy4 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 		phy-speed = <COMPHY_SPEED_10_3125G>;
 	};
 	phy5 {
@@ -298,7 +298,7 @@
 		phy-type = <COMPHY_TYPE_UNCONNECTED>;
 	};
 	phy4 {
-		phy-type = <COMPHY_TYPE_SFI>;
+		phy-type = <COMPHY_TYPE_SFI0>;
 		phy-speed = <COMPHY_SPEED_10_3125G>;
 	};
 	phy5 {
diff --git a/drivers/phy/marvell/comphy_core.c b/drivers/phy/marvell/comphy_core.c
index cb59788277..c849166d48 100644
--- a/drivers/phy/marvell/comphy_core.c
+++ b/drivers/phy/marvell/comphy_core.c
@@ -37,8 +37,8 @@ static char *get_type_string(u32 type)
 		"UNCONNECTED", "PEX0", "PEX1", "PEX2", "PEX3",
 		"SATA0", "SATA1", "SGMII0", "SGMII1", "SGMII2",
 		"USB3", "USB3_HOST0", "USB3_HOST1",
-		"USB3_DEVICE", "RXAUI0", "RXAUI1", "SFI", "AP",
-		"IGNORE"};
+		"USB3_DEVICE", "RXAUI0", "RXAUI1", "SFI0", "SFI1",
+		"AP", "IGNORE"};
 
 	if (type < 0 || type > COMPHY_TYPE_MAX)
 		return "invalid";
diff --git a/drivers/phy/marvell/comphy_cp110.c b/drivers/phy/marvell/comphy_cp110.c
index 156b0a6163..07ff0c15de 100644
--- a/drivers/phy/marvell/comphy_cp110.c
+++ b/drivers/phy/marvell/comphy_cp110.c
@@ -196,10 +196,11 @@ int comphy_cp110_sfi_rx_training(struct chip_serdes_phy_config *ptr_chip_cfg,
 				 u32 lane)
 {
 	int ret;
+	u32 type = ptr_chip_cfg->comphy_map_data[lane].type;
 
 	debug_enter();
 
-	if (ptr_chip_cfg->comphy_map_data[lane].type != COMPHY_TYPE_SFI) {
+	if (type != COMPHY_TYPE_SFI0 && type != COMPHY_TYPE_SFI1) {
 		pr_err("Comphy %d isn't configured to SFI\n", lane);
 		return 0;
 	}
@@ -603,13 +604,14 @@ int comphy_cp110_init(struct chip_serdes_phy_config *ptr_chip_cfg,
 					 ptr_chip_cfg->comphy_base_addr, lane,
 					 mode);
 			break;
-		case COMPHY_TYPE_SFI:
-			mode = COMPHY_FW_FORMAT(COMPHY_SFI_MODE,
-						COMPHY_UNIT_ID0,
+		case COMPHY_TYPE_SFI0:
+		case COMPHY_TYPE_SFI1:
+			/* Calculate SFI id */
+			id = ptr_comphy_map->type - COMPHY_TYPE_SFI0;
+			mode = COMPHY_FW_FORMAT(COMPHY_SFI_MODE, id,
 						ptr_comphy_map->speed);
 			ret = comphy_smc(MV_SIP_COMPHY_POWER_ON,
-					 ptr_chip_cfg->comphy_base_addr, lane,
-					 mode);
+				ptr_chip_cfg->comphy_base_addr, lane, mode);
 			break;
 		case COMPHY_TYPE_RXAUI0:
 		case COMPHY_TYPE_RXAUI1:
diff --git a/include/dt-bindings/comphy/comphy_data.h b/include/dt-bindings/comphy/comphy_data.h
index d46fc886fd..7736bb07b7 100644
--- a/include/dt-bindings/comphy/comphy_data.h
+++ b/include/dt-bindings/comphy/comphy_data.h
@@ -33,10 +33,11 @@
 #define COMPHY_TYPE_USB3_DEVICE		13
 #define COMPHY_TYPE_RXAUI0		14
 #define COMPHY_TYPE_RXAUI1		15
-#define COMPHY_TYPE_SFI			16
-#define COMPHY_TYPE_AP			17
-#define COMPHY_TYPE_IGNORE		18
-#define COMPHY_TYPE_MAX			19
+#define COMPHY_TYPE_SFI0		16
+#define COMPHY_TYPE_SFI1		17
+#define COMPHY_TYPE_AP			18
+#define COMPHY_TYPE_IGNORE		19
+#define COMPHY_TYPE_MAX			20
 #define COMPHY_TYPE_INVALID		0xff
 
 #define COMPHY_POLARITY_NO_INVERT	0
-- 
2.22.0

